commit 8103a191ec07ab0380be65b67128160043bc19d2
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2019-05-15T22:40:49+00:00 (5 months ago)
    
    joyent/node-lockfd#8 Async shared file locks test fails with node v4

diff --git a/lib/index.js b/lib/index.js
index de2d58b..b00c7c0 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -22,15 +22,14 @@ lockfd(fd, callback)
 	check_arg(2, 'callback', callback, 'function');
 
 	BINDING.lock_fd(fd, 'write', false, function (ret, errmsg, errno) {
+		var err = null;
+
 		if (ret === -1) {
-			var err = new Error('File Locking Error: ' + errmsg);
+			err = new Error('File Locking Error: ' + errmsg);
 			err.code = errno;
-
-			callback(err);
-			return;
 		}
 
-		callback(null);
+		setImmediate(callback, err);
 	});
 }
 
@@ -69,15 +68,14 @@ flock(fd, op, callback)
 	check_arg(3, 'callback', callback, 'function');
 
 	BINDING.flock(fd, op, false, function (ret, errmsg, errno) {
+		var err = null;
+
 		if (ret === -1) {
-			var err = new Error('File Locking Error: ' + errmsg);
+			err = new Error('File Locking Error: ' + errmsg);
 			err.code = errno;
-
-			callback(err);
-			return;
 		}
 
-		callback(null);
+		setImmediate(callback, err);
 	});
 }
 
diff --git a/package.json b/package.json
index 6468bcb..9c4b9c1 100644
--- a/package.json
+++ b/package.json
@@ -1,12 +1,12 @@
 {
   "name": "lockfd",
   "description": "thin wrapper around fcntl(F_SETLK) and flock()",
-  "version": "2.0.1",
+  "version": "2.0.2",
   "author": "Joshua M. Clulow <jmc@joyent.com>",
   "repository": "git://github.com/joyent/node-lockfd.git",
   "main": "./lib/index.js",
   "dependencies": {
-    "v8plus": "~1.0.2"
+    "v8plus": "~1.0.3"
   },
   "scripts": {
     "postinstall": "cd src && gmake $(eval echo ${MAKE_OVERRIDES})",
diff --git a/test/unit/flock.test.js b/test/unit/flock.test.js
index 5c8b7ee..e81713f 100644
--- a/test/unit/flock.test.js
+++ b/test/unit/flock.test.js
@@ -38,13 +38,13 @@ function asyncTestRun(t, path, ltype, callback) {
 
 	function getSecondLock() {
 		mod_lockfd.flock(fd2, ltype, function (err) {
-			t.ifErr(err, 'acquired lock successfully');
+			t.ifErr(err, 'workload 2 acquired lock successfully');
 			doWork(fd2, 'workload 2');
 		});
 	}
 
 	mod_lockfd.flock(fd1, ltype, function (err) {
-		t.ifErr(err, 'acquired lock successfully');
+		t.ifErr(err, 'workload 1 acquired lock successfully');
 		process.nextTick(getSecondLock);
 		setTimeout(doWork, TIMEOUT, fd1, 'workload 1');
 	});
